tests/unit/test_synthDrivers/test_espeak.py

```diff
@@ -0,0 +1,83 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2022 NV Access Limited.
+
+"""Unit tests for the eSpeak synth driver submodule.
+"""
+
+import logging
+import unittest
+import logHandler
+from speech.commands import LangChangeCommand
+from source.synthDrivers.espeak import SynthDriver
+
+
+class FakeESpeakSynthDriver:
+	_language = "default"
+	_defaultLangToLocale = {"default": "en-gb"}
+	availableLanguages = {"fr", "fr-fr", "en-gb", "ta-ta"}
+
+
+class TestSynthDriver(unittest.TestCase):
+	def setUp(self) -> None:
+		self._driver = SynthDriver()
+	
+	def tearDown(self) -> None:
+		self._driver.terminate()
+
+	def test_determineLangFromCommand(self):
+		"""Test cases for determining a supported eSpeak language from a LangChangeCommand."""
+		self.assertEqual(
+			"en-gb",
+			SynthDriver._determineLangFromCommand(FakeESpeakSynthDriver, LangChangeCommand(None)),
+			msg="Default language used if language code not provided"
+		)
+		self.assertEqual(
+			"fr-fr",
+			SynthDriver._determineLangFromCommand(FakeESpeakSynthDriver, LangChangeCommand("fr_FR")),
+			msg="Language with locale used when available"
+		)
+		self.assertEqual(
+			"en-gb",
+			SynthDriver._determineLangFromCommand(FakeESpeakSynthDriver, LangChangeCommand("default")),
+			msg="Default eSpeak language mappings used"
+		)
+		self.assertEqual(
+			"fr",
+			SynthDriver._determineLangFromCommand(FakeESpeakSynthDriver, LangChangeCommand("fr_FAKE")),
+			msg="Language without locale used when available"
+		)
+		self.assertEqual(
+			"ta-ta",
+			SynthDriver._determineLangFromCommand(FakeESpeakSynthDriver, LangChangeCommand("ta-gb")),
+			msg="Language with any locale used when available"
+		)
+		with self.assertLogs(logHandler.log, level=logging.DEBUG) as logContext:
+			self.assertEqual(
+				None,
+				SynthDriver._determineLangFromCommand(FakeESpeakSynthDriver, LangChangeCommand("fake")),
+				msg="No matching available language returns None"
+			)
+		self.assertIn(
+			"Unable to find an eSpeak language for 'fake'",
+			logContext.output[0]
+		)
+
+	def test_defaultMappingAvailableLanguage(self):
+		"""Confirms language codes remapped by default are supported by eSpeak via integration testing"""
+		mappedDefaultLanguages = set(self._driver._defaultLangToLocale.values())
+		unexpectedUnsupportedDefaultLanguages = mappedDefaultLanguages.difference(self._driver.availableLanguages)

```

Comment:
Could you add the full output of `self._driver.availableLanguages` on this PR (in a collapsed section comment) I think it will help to add context.

Reply:
Where should this be added? This is a 100 line comment. 
I was thinking it should be added to `synthDriver.espeak.SynthDriver.availableLanguages`.

Without a pretty print:
```py
# For eSpeak commit 7e5457f91e10 (NVDA 2022.1), availableLanguages resolves to:
{'ia', 'ru', 'cy', 'ms', 'af', 'fi', 'fr-fr', 'nog', 'gu', 'hu', 'eu', 'om', 'en-029', 'de', 'es', 'kk', 'an', 'nci', 'uk', 'vi-vn-x-south', 'grc', 'it', 'vi-vn-x-central', 'bg', 'piqd', 'ug', 'ar', 'da', 'mi', 'mr', 'pt-br', 'fr-ch', 'py', 'uz', 'en-gb', 'sw', 'as', 'shn', 'vi', 'nl', 'bs', 'ga', 'pap', 'sv', 'kn', 'gn', 'th', 'tr', 'pa', 'mt', 'chr-US-Qaaa-x-west', 'eo', 'kok', 'ky', 'lfn', 'is', 'pt', 'en-gb-x-gbcwmd', 'en-gb-x-rp', 'ht', 'bpy', 'fr-be', 'nb', 'lt', 'ja', 'te', 'tn', 'es-419', 'gd', 'sjn', 'he', 'hyw', 'et', 'ro', 'ru-lv', 'sq', 'quc', 'am', 'hr', 'qya', 'ka', 'el', 'tt', 'or', 'pl', 'qu', 'ba', 'ta', 'cmn', 'io', 'en-us', 'ur', 'hi', 'en-gb-scotland', 'fa', 'kl', 'tk', 'ku', 'si', 'cv', 'ca', 'qdb', 'hak', 'fa-latn', 'lv', 'en-gb-x-gbclan', 'ltg', 'ne', 'sl', 'az', 'yue', 'sk', 'hy', 'my', 'ko', 'mk', 'smj', 'ml', 'cmn-latn-pinyin', 'id', 'la', 'sr', 'bn', 'sd', 'cs', 'jbo', 'haw'}
"""